tests/rptest/tests/partition_movement_test.py

```diff
@@ -30,7 +30,9 @@
     # e.g.  raft - [follower: {id: {1}, revision: {10}}] [group_id:3, {kafka/topic/2}] - recovery_stm.cc:422 - recovery append entries error: raft group does not exists on target broker
     "raft - .*raft group does not exist on target broker",
     # e.g.  raft - [group_id:3, {kafka/topic/2}] consensus.cc:2317 - unable to replicate updated configuration: raft::errc::replicated_entry_truncated
-    "raft - .*unable to replicate updated configuration: .*"
+    "raft - .*unable to replicate updated configuration: .*",
+    # e.g. recovery_stm.cc:432 - recovery append entries error: rpc::errc::client_request_timeout"
+    "raft - .*recovery append entries error.*client_request_timeout"
```

Comment:
I'm aiming to scope the allow lists as tightly as possible -- ignoring specific types of error rather than all possible errors from a particular LOC.  It's a bit of a balancing act, but hopefully helpful to someone looking at this future during log cleanup: they can see from our allow list which particular error types triggered this error path.